src/core/network/qgshttpheaders.h

```diff
@@ -126,13 +126,26 @@ class CORE_EXPORT QgsHttpHeaders
     bool updateMap( QVariantMap &map ) const;
 
     /**
-     * \brief Updates a \a map by adding all the HTTP headers
+     * \brief Updates a DOM element by adding all the HTTP headers
+     *
+     * KEY_REFERER value will be available at attribute "KEY_PREFIX+KEY_REFERER" and attribute "KEY_REFERER" (for backward compatibility)
+     *
+     * \param el DOM element
+     * \return TRUE if the update succeed
+     * \deprecated QGIS 3.42. Use updateDomElement2() const instead.
+     */
+    Q_DECL_DEPRECATED bool updateDomElement( QDomElement &el ) const;
+
+    /**
+     * \brief Updates a DOM element by adding all the HTTP headers
      *
      * KEY_REFERER value will be available at attribute "KEY_PREFIX+KEY_REFERER" and attribute "KEY_REFERER" (for backward compatibility)
      *
+     * \param el DOM element
+     * \param[out] namespaceDeclarations Map of (prefix, URI) tuples for namespaces used by el.
      * \return TRUE if the update succeed
      */
-    bool updateDomElement( QDomElement &el ) const;
+    bool updateDomElement2( QDomElement &el, QMap<QString, QString> &namespaceDeclarations SIP_OUT ) const;
```

Comment:
The naming with a '2' suffix is obviously a bit silly, but we can't have an override with the same name and a out parameter without breaking the Python API

Reply:
One could argue this stuff is probably not very useful in the Python API...

Reply:
Why don't we make the new version private, and then just mark the old as deprecated "will be removed in QGIS 4"?

Reply:
done